### ISSUE TITLE
Deconstructed variables code fix should always camelCase variables

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#23716]
``` C#
class Program
{
    private static readonly (string Factory, string EvaluationRuleName) Foo;

    static void Main(string[] args)
    {
        var foo = Foo; // Deconstruct tuple here


    }
}
```

Expected:
``` C#
class Program
{
    private static readonly (string Factory, string EvaluationRuleName) Foo;

    static void Main(string[] args)
    {
        var (factory, evaluationRuleName) = Foo;
    }
}
```

Actual:

``` C#

class Program
{
    private static readonly (string Factory, string EvaluationRuleName) Foo;

    static void Main(string[] args)
    {
        var (Factory, EvaluationRuleName) = Foo;

    }
}
```

Because tuple names appear as public fields when you access them via the tuple itself, it's common practice to `PascalCase` them in a declaration. However, when they are deconstructed into variables - you want them them to be `camelCase` to match the naming style for variables.

Either add an option here that I can switch or change the behavior. As it stands, I can't use this code fix.

**Comment by sharwell** [comment]
> ... tuple names appear as public fields ...

Technically, the names are metadata and not actually fields. The field names are just `Item1` and `Item2`.

> it's common practice to PascalCase them in a declaration

I can see how this approach makes sense, but so far I'm not sure we've seen enough cases to say what is common practice and what is not.

With that said, the proposed approach sounds reasonable. The part that's less clear is how well this will hold up in other related scenarios (i.e. other cases where declared tuple names differ from local variable naming conventions).

**Comment by sharwell** [comment]
@zaytsev-victor If you want to proceed with this, it's likely to be accepted unless you found a clear case where it wouldn't work (and thus needs further design work)

**Comment by CyrusNajmabadi** [comment]
LDM guidance is that tuple fields should be named with camelCasing, not pascal casing.  Precisely because they're for shuttling around a few variables.  That they're stored as 'Fields' in a 'Tuple' type is an implementation detail. You're not supposed to think of them as fields, or otherwise treat them that way.

**Comment by CyrusNajmabadi** [comment]
A strong argument that was made that caused the LDM to go this way was the following.

If i have:

```c#
void MyMethod(int x, int y, int z, int length) { }
```

And i convert that to:

```c#
void MyMethod((int, int, int) v, int length) { }
```

What should the names be?  There was a strong feeling that people should feel that a grouping of parameters into a tuple feels and are named the same as the expanded form.  So that would be written as:

```c#
void MyMethod((int x, int y, int z) v, int length) { }
```

And not


```c#
void MyMethod((int X, int Y, int Z) v, int length) { }
```

**Comment by CyrusNajmabadi** [comment]
in other words: Tuples are like parameter lists.  You camel case parameters.  So you camel case tuple member names.

**Comment by davkean** [comment]
I agree with inputs - and that it definitely feels natural. I'm not yet convinced on the outputs:

``` C#
        static void Main(string[] args)
        {
            var point = GetPoint();

            Console.WriteLine(point.x);

        }

        private static (int x, int y) GetPoint()
        {
            return default((int, int));
        }
```

For intent and purposes, it looks like a field:
![image](https://user-images.githubusercontent.com/1103906/33856050-fb05f440-df1a-11e7-93a8-9ab0e7810c48.png)

**Comment by davkean** [comment]
But, I can see your argument - inputs are deconstructed just like outputs.

**Comment by CyrusNajmabadi** [comment]
We'd take a community pr here.
